impl/src/main/java/com/okta/sdk/impl/ds/DefaultDataStore.java

```diff
@@ -595,4 +596,13 @@ private HttpHeaders headers(Map<String, List<String>> headerParameters) {
     public RequestBuilder http() {
         return new DefaultRequestBuilder(this);
     }
+
+    @Override
+    public boolean isReady() {
+        try {
+            return getRawResponse("/api/v1/apps", null, null) != null;
+        } catch (ResourceException | HttpException exception) {
+            return false;
```

Comment:
this exception should be logged, probably at debug.

I'd also question the use of the apps API here, In the case of OAuth tokens, the token might not be scoped to be able to use this API, which may cause a false positive. cc: @arvindkrishnakumar-okta

Reply:
In that case it would be difficult to troubleshoot as the exception isn't logged ;)

Reply:
I guess the need here is to just ensure that the connectivity between client and Okta Org is fine. Assuming this, a simple `ping` functionality should suffice here instead of a fully qualified API URL to hit?

Reply:
I think the need here is to just ensure that the connectivity between client and Okta Org is fine. We could probably hit the OIDC discovery endpoint `https://{yourOktaDomain}/.well-known/openid-configuration` and determine if the client-server communication channel is ready.

Reply:
if the goal is to see that the client is ready (and can authenticate). The well-known endpoint might not work as it's anonymous.

I'm not sure this feature could reliably work for OAuth clients (given the potential scope issues).  You can also scope an SSWS token, so there are some potential issues there too 🤔 

Maybe this is a feature we recommend folks implement on their own? and provide some sort of doc?
basically what @VitaliiTytarenko-okta originally had, but in a README instead?

```java
 try {
    client.listApplications()
} catch (ResourceException exception) {
    // handle client not ready state
}
```

It's ugly, or we could also do something with a lambda:

```java
boolean ready = client.isReady(() -> client.listApplications())
```

I don't think it's obvious what that does, but.. we could also take the highbred approach and default `isReady()` to the apps endpoint as well.
While a bit more elegant than the first, it comes with the cost of complicated docs. It's also not something a client _should_ regularly do except in the case of some sort to initialization loop, which might already be custom logic?  🤔   

Thoughts?

Reply:
We can possibly do something like `isReady("client-specified-url)"` instead of a preset `list apps` URL?

Reply:
Unfortunately, I haven't come up with anything better. I don't like an idea to pass URL as a parameter, because any URL can be passed and we can't guarantee anything. So I used a method reference as a `Supplier<? extends Resource>`.